packages/ai/src/AiSlice.ts

```diff
@@ -83,7 +83,7 @@ export function createAiSlice<PC extends BaseRoomConfig & AiSliceConfig>({
   customTools = {},
   getInstructions,
 }: {
-  getApiKey: (modelProvider: string) => string;
+  getApiKey?: (modelProvider: string) => string;
```

Comment:
I think it'd be good to have baseUrl as one of the props. Could we add a union type here? Something like:

```ts
 {... /* rest of the props */ } &
 ({baseUrl: string}  |
  {getApiKey: (modelProvider: string) => string;})
```